src/build/incrementality.go

```diff
@@ -211,9 +211,7 @@ func ruleHash(state *core.BuildState, target *core.BuildTarget, runtime bool) []
 	for _, licence := range target.Licences {
 		h.Write([]byte(licence))
 	}
-	for _, output := range target.TestOutputs {
-		h.Write([]byte(output))
-	}
```

Comment:
Pretty sure these should only contribute to the runtime hash.

Reply:
> Makes sense.
> 
> Note that these changes will mean it will rebuild _everything_. Maybe we just live with that though.

I mean we do that quite a bit anyway because the tools are in the root of the build graph. Might want to think about this because it was better in this sense before we moved tools to remote files.

Reply:
Yeah. I still think breaking out the tools to separate repos (plugins!) would be the best solution there.